File: 프로그래머스/LV0/코딩테스트 입문/합성수 찾기/합성수 찾기.py
def count_composite_number(n):
    sieve = [True] * (n+1)

    m = int(n ** 0.5)
    for i in range(2, m + 1):
        if sieve[i] == True:
            for j in range(i+i, n+1, i):
                sieve[j] = False
    return sieve.count(False)

def solution(n):
    return count_composite_number(n)

File: 프로그래머스/LV0/코딩테스트 입문/합성수 찾기/합성수 찾기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>

int count_composite_number(n)
{
    int ret = 0;
    bool* sieve = malloc(n + 1);
    memset(sieve, true, n + 1);

    int m = (int)pow(n, 0.5);

    for (int i = 2; i <= m; i++)
    {
        if (sieve[i] == true)
        {
            for (int j = 2 * i; j <= n; j += i)
            {
                sieve[j] = false;
            }
        }
    }

    for (int i = 2; i <= n; i++)
    {
        if (!sieve[i])ret++;
    }

    return ret;
}

int solution(int n) 
{
    return count_composite_number(n);
}